python/arcticdb/version_store/_store.py

```diff
@@ -956,6 +956,7 @@ def batch_read(
             all(v is not None for v in versioned_items),
             "Null value from _batch_read_to_versioned_items. NoDataFoundException should have been thrown instead.",
         )
+        self.version_store.trim()
```

Comment:
I don't know how much difference trim() will make here, as the user will still be holding references to the allocated results. We should probably remove it here and add something to the documentation telling them that they can call trim themselves once they've released all references to the returned dataframes and done a gc.collect() to remove any reference cycles

Reply:
I'm a bit confused what benefit the malloc_trim is giving us here. On your write up on the ticket @G-D-Petrov you write,

```
Removing malloc_trim doesn't seem to have affected the memory usage.
Even on my stress test that went from 1000 up to 60000 symbols, in a continuous loop, there were no noticeable memory issues.
```

Shall we just get rid of it completely?

I'm a bit hesistant to put such complicated semantics: 

```
they can call trim themselves once they've released all references to the returned dataframes and done a gc.collect() to remove any reference cycles
```

on the public API (and commit to supporting them). Especially considering our target users are statisticians as much as developers.

Reply:
William explained the history lesson to me about why the trim is there.

Still keen not to put it on the external API `Library`. Makes it hard to change our minds about it.

How about we keep it on the `NativeVersionStore` only for now? Then `Library` users can still use it in an emergency via the `_nvs` handle.

Reply:
Okay I will remove the method from the Library API.
Do we want to keep the documentation that I added about this?
And point people that they can use the _nvs handle?